cf-agent/verify_files_utils.c

```diff
@@ -854,7 +863,16 @@ static PromiseResult SourceSearchAndCopy(EvalContext *ctx, const char *from, cha
             if (cf_stat(newfrom, &sb, attr.copy, conn) == -1)
             {
                 Log(LOG_LEVEL_VERBOSE, "Can't stat '%s'. (cf_stat: %s)", newfrom, GetErrorStr());
-                continue;                                       /* TODO FAIL? */
+                if (conn->conn_info->is_timeouted)
+                {
+                    cfPS(ctx, LOG_LEVEL_INFO, PROMISE_RESULT_INTERRUPTED, pp,
+                         attr, "connection timeout");
+                    return PROMISE_RESULT_INTERRUPTED;
```

Comment:
@jimis this is not true unfortunately. I agree we should close connection if there is any problem with it detected so I'll change this accordingly.
Unfortunately this won't eliminate the mess. The connection problems are discovered "deep inside network path" and this stuff is called by many functions in many places. In order to fix whole call stuck you will have to probably rewrite fair amount of functions and analyze which are supposed to close the connection and which are not.
Setting a flag and checking it when needed it IMHO much cleaner approach and in fact less risky. We will set flag in correct place and check it exactly where we are deciding if we should continue or not.